### PR TITLE
feat(tools): VM-194 performance metering

### PR DESCRIPTION
**Pull request by monoid**

### CODE REVIEW

File: tools/cli/performance_metering/setup.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#
+from setuptools import setup
+
+
+setup(name='aquavm_performance_metering',
+      version='0.1',
+      description='An AquaVM Performance metering tool',
+      author='Fluence Labs',
+      author_email='devs@fluencelabs.one',
+      license='Apache',
```
Comment (monoid) [inline tools/cli/performance_metering/setup.py:24]: Apache-2?

File: junk/cidify/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "cidify"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (mikevoronov) [inline junk/cidify/Cargo.toml:6]: let's remove it

File: benches/PERFORMANCE.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        "from_success_result": "0:00:00.009959",
+        "make_exec_ctx": "0:00:00.002729",
+        "parse": "0:00:01.358000",
+        "populate_outcome_from_contexts": "0:00:00.007437",
+        "prepare": "0:00:01.372000"
+      }
+    },
+    "version": "0.35.0"
+  }
+}
```
Comment (mikevoronov) [inline benches/PERFORMANCE.json:35]: suggest add a newline just not to see this sign in every commit when this file is updated

**Review by mikevoronov — APPROVED** [review]
utACK

File: benches/PERFORMANCE.json [hunk 4]
```diff
@@ -0,0 +1,302 @@
+{
+  "d77ebe8481884bc3b2778c8083f1bf459e548e929edd87041beb14f6b868d35f": {
+    "benches": {
+      "big_values_data": {
+        "comment": "Loading a trace with huge values",
+        "stats": {
+          "air::farewell_step::outcome::from_success_result": {
```
Comment (mikevoronov) [inline benches/PERFORMANCE.json:7]: Suggest to delete `air::` prefix because it's common

**Review by mikevoronov — CHANGES_REQUESTED** [review]

File: benches/PERFORMANCE.json (hunk 4 above)
Comment (monoid) [reply benches/PERFORMANCE.json:7]: I calculate nested functions' common prefix and remove it in each node.

File: benches/PERFORMANCE.txt [hunk 5]
```diff
@@ -0,0 +1,77 @@
+Machine d77ebe8481884bc3b2778c8083f1bf459e548e929edd87041beb14f6b868d35f:
+  Platform: macOS-13.1-arm64-arm-64bit
+  Timestamp: 2023-02-02 17:09:14.780857+00:00
+  AquaVM version: 0.35.0
+  Benches:
+    big_values_data: Loading a trace with huge values
```
Comment (mikevoronov) [inline benches/PERFORMANCE.txt:6]: let's write the whole time after a test name

File: benches/PERFORMANCE.txt [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      air::preparation_step::preparation::prepare: 310.00ms
+        air_interpreter_data::interpreter_data::serde_json::from_slice: 292.40ms
+        air_parser::parser::air_parser::parse: 3.73ms
+        air::preparation_step::preparation::make_exec_ctx: 3.39ms
+      air::runner::execute: 250.00µs
+      air::farewell_step::outcome::from_success_result: 29.03ms
+        populate_outcome_from_contexts: 26.65ms
+          to_vec(data): 21.49ms
+          to_vec(call_results): 182.00µs
+    network_explore: network_explore-inspired script
```
Comment (mikevoronov) [inline benches/PERFORMANCE.txt:48]: let's just write smth like "a test where a N peers in a network are discovered" as a comment for this test

**Review by mikevoronov — APPROVED** [review]
utACK

**Pull request merged**